### PR TITLE
Add some documentation for the format of meta/runtime.yml

### LINKED ISSUE #70918: Add /meta and runtime.yml docs to Developer Guide
##### SUMMARY
Looking at the Developer Guide (devel branch):
https://docs.ansible.com/ansible/devel/dev_guide/developing_collections.html

Not seeing any mention of `runtime.yml` nor the `/meta` directory as part of the schema.

##### ISSUE TYPE
- Documentation Report

##### COMPONENT NAME
Unsure

##### ANSIBLE VERSION
Devel - 2.10

##### CONFIGURATION
N/A

##### OS / ENVIRONMENT
N/A

##### ADDITIONAL INFORMATION
Developers should be using semver as well as providing all info in the `runtime.yml` file (like `requires_ansible` and more).

### PR DESCRIPTION
**Pull request by s-hertel**
##### SUMMARY
Fixes #70918

##### ISSUE TYPE
- Docs Pull Request

### CODE REVIEW

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 1]
```diff
@@ -194,6 +196,60 @@ command completion, or environment variables are presented throughout the
...
+.. _meta_runtime_yml:
+
+meta directory
+--------------
+
+A collection can store some additional metadata in a ``runtime.yml`` file in the collection's ``meta`` directory. The ``runtime.yml`` file supports the top level keys:
+
+- *requires_ansible*:
+
+  The version of Ansible required to use the collection.
```
Comment (jborean93) [inline docs/docsite/rst/dev_guide/developing_collections.rst]: Would be a good idea to mention you can specify multiple versions separated by a comma, e.g. `requires_ansible: '>=2.10,<2.11'`

File: docs/docsite/rst/dev_guide/developing_collections.rst [hunk 2]
```diff
@@ -194,6 +196,60 @@ command completion, or environment variables are presented throughout the
...
+
+  .. code:: yaml
+
+     requires_ansible: ">=2.10"
+
+- *plugin_routing*:
+
+  A mapping of content in a collection that has been relocated or deprecated.
+  The top level keys of ``plugin_routing`` are types of plugins (for example, ``become``, ``connection``, and ``inventory``).
+  The individual resources under those types can define ``redirect`` for a new location and ``deprecation`` and ``tombstone`` to define a warning message and removal date or version.
```
Comment (jborean93) [inline docs/docsite/rst/dev_guide/developing_collections.rst]: Is there a difference between `deprecation` and `tombstone`?

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 1 above)
Comment (s-hertel) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: That's a good idea

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 2 above)
Comment (s-hertel) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: I believe `tombstone` is intended to be used when a plugin is actually removed after the deprecation period. (@nitzmahone should probably confirm that though)

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 1 above)
Comment (nitzmahone) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: We should probably mention somewhere that this is a PEP440 Version Specifier set under the covers (https://www.python.org/dev/peps/pep-0440/#version-specifiers), and that we deviate from the PEP440 behavior by truncating prerelease segments from the Ansible version before running it through the compatibility sieve (so eg, Ansible 2.11.0b1 is compatible with something that lists >=2.11, even thought PEP440 says it's not).

File: docs/docsite/rst/dev_guide/developing_collections.rst (hunk 2 above)
Comment (nitzmahone) [reply docs/docsite/rst/dev_guide/developing_collections.rst]: `deprecation` generates a runtime warning (unless disabled by `DEPRECATION_WARNINGS` config) about plugins and/or `redirect`s that will be removed in future releases. eg, a routing entry that contains both `deprecation` and `redirect` will issue a warning when the plugin is accessed via that routing entry, then send it along to the redirected location. `tombstone` generates a fatal error for plugins that have already been removed. Plugins that have merely been moved elsewhere should use `redirect` not `tombstone`, as it's a fatal error; it should only be used for a plugin that's truly dead (eg, during collections search, we don't keep looking if we hit a `tombstone` entry).

**Review by jborean93 — APPROVED** [review]

**Review by samccann — APPROVED** [review]
LGTM

**Comment by nitzmahone** [comment]
Thanks for adding this @s-hertel !

**Pull request merged**